web/app/build/build.component.ts

```diff
@@ -12,20 +13,29 @@ import {BuildLogWebsocketService} from '../services/build-log-websocket.service'
 })
 export class BuildComponent implements OnInit {
   logs = '';
+  readonly breadcrumbs: Breadcrumb[] =
+      [{label: 'Dashboard', url: '/'}, {hint: 'Project'}, {hint: 'Build'}];
 
   constructor(
       private readonly buildLogSocketService: BuildLogWebsocketService,
       private readonly route: ActivatedRoute) {}
 
   ngOnInit() {
     this.route.paramMap
-        .switchMap(
-            (params: ParamMap) => this.buildLogSocketService.connect(
-                params.get('projectId'), params.get('buildId')))
+        .switchMap((params: ParamMap) => {
+          this.updateBreadcrumbs(params.get('projectId'));
+          return this.buildLogSocketService.connect(
+              params.get('projectId'), params.get('buildId'));
+        })
         .subscribe((message) => {
           // TODO: this will be more than a string later on. Let's define a log
           // line model.
           this.logs += message.data;
         });
   }
+
+  private updateBreadcrumbs(projectId: string): void {
+    // TODO: update the breadcrumbs to have Project name and Build number
+    this.breadcrumbs[1].url = `/project/${projectId}`;
```

Comment:
Is the suggestion that `references` will end up providing the route definitions to the web app?
I think the routes should be static in the web app, it shouldn't be determined by the backend. I think what we have where the backend builds the URL is correct. One place where it might get tricky is we ever have the frontend being hosted separately from the backend and the host names differ.

Reply:
Yeah, that's what I meant. What do you think about the above mentioned example though, where we have to report the build URL to the GitHub status API for the `Details` button next to a build status?

Reply:
Yes! I think we don't have to include the host name in the URL and let the "user" of the `build_url` method add it.

e.g.

- The web frontend needs the build URL to link to it in a list: `/project/123/build/123` works, as the browser automatically deals with the host name
- The sinatra server needs the build URL to link to it on a GitHub status page: it will add the host name right before sending the URL to GitHub. No need to change the `Details` page of a PR in retro in case the host name changes as you pointed out.

What do you think?

Reply:
>The web frontend needs the build URL to link to it in a list: /project/123/build/123 works, as the browser automatically deals with the host name

I think the frontend should be in control of this. It is the one who defines the route, would be repetitive to make requests for every build URL. Other than logic, the additional http requests would be extra latency that we would have to handle in the frontend as well.

>The sinatra server needs the build URL to link to it on a GitHub status page: it will add the host name right before sending the URL to GitHub. No need to change the Details page of a PR in retro in case the host name changes as you pointed out.

This makes sense 👍